docs/development/best-practices.md

```diff
@@ -0,0 +1,112 @@
+# Best Practices
+
+## General
+
+- Prefer full function declaration, as opposed to `const func = ():void => {}`, for readability and better stack traces
+- Use logger metadata:
+
+Use
+
+```js
+logger.trace({ obj }, 'title');
+```
+
+instead of
+
+```js
+logger.trace('title:\n' + obj);
+```
+
+- Use descriptive `istanbul` comments
```

Comment:
Also prevent `istanbul` comments and add tests. Only use those comments if codepath is required for coimpilations but never hit at runtime.